pkg/api/blocks/v1.go

```diff
@@ -72,8 +73,9 @@ func NewBlocksAPI(logger log.Logger, disableCORS bool, label string, flagsMap ma
 			Blocks: []metadata.Meta{},
 			Label:  label,
 		},
-		disableCORS: disableCORS,
-		bkt:         bkt,
+		disableCORS:            disableCORS,
+		bkt:                    bkt,
+		disableAdminOperations: flagsMap["disable-admin-operations"] == "true",
```

Comment:
actaully `flagsMap map[string]string` will give a string "true" with `flagsMap["disable-admin-operations"]`. thus I am simply converting it to boolean.

Reply:
It would be cleaner and more consistent with other such options like disableCors, and we wouldn't need to check a string here at all
But actually, I think we don't even need any changes in BlocksAPI here. The flag value will just directly be used in UI anyway, so we don't need a separate API. See my other comment https://github.com/thanos-io/thanos/pull/6601#discussion_r1289653181

Reply:
okay! thanks for the review. i will look into it!

Reply:
> It would be cleaner and more consistent with other such options like disableCors, and we wouldn't need to check a string here at all
> But actually, I think we don't even need any changes in BlocksAPI here. The flag value will just directly be used in UI anyway, so we don't need a separate API. See my other comment [#6601 (comment)](https://github.com/thanos-io/thanos/pull/6601#discussion_r1289653181)

@saswatamcode one problem that I am facing when using `/status/flags` endpoint is that I am not able to figure out how to get the flags value inside the markBlock function of v1.go. I need it so as to restrict access to the API when the flag is enabled.